### ISSUE TITLE
Can't compile

### ISSUE THREAD

**Issue by oz9ny** [K7MDL2/KEITHSDR#3]
Hi 
I have tried to compile this project with no success.

My setup: Win10, Arduino 1.8.16, Teensyduino 1.55

Below is the error messages from compilation:

73 de Niels, OZ9NY

Arduino: 1.8.16 (Windows 10), TD: 1.55, Board: "Teensy 4.1, Serial, 600 MHz, Faster, US English"

SDR_CAT.cpp:64: warning: 'msg' defined but not used 
 static char msg[S_BUFF];
             ^

SD_Card.cpp: In function 'void SD_CardInfo()':
SD_Card.cpp:177: error: no matching function for call to 'SdFile::openRoot(SdVolume&)'
     root.openRoot(volume);
                         ^
In file included from C:\Program Files (x86)\Arduino\hardware\teensy\avr\libraries\SdFat\src/FsLib/FsLib.h:32:0,
                 from C:\Program Files (x86)\Arduino\hardware\teensy\avr\libraries\SdFat\src/SdFat.h:35,
                 from C:\Program Files (x86)\Arduino\hardware\teensy\avr\libraries\SD\src/SD.h:27,
                 from C:\Users\Niels\Documents\hamradio\KEITHSDR\SDR_RA8875\SDR_RA8875.h:19,
                 from C:\Users\Niels\Documents\hamradio\KEITHSDR\SDR_RA8875\SD_Card.cpp:7:

C:\Program Files (x86)\Arduino\hardware\teensy\avr\libraries\SdFat\src/FsLib/FsFile.h:458:8: note: candidate: bool FsBaseFile::openRoot(FsVolume*)
   bool openRoot(FsVolume* vol);
        ^
C:\Program Files (x86)\Arduino\hardware\teensy\avr\libraries\SdFat\src/FsLib/FsFile.h:458:8: note:   no known conversion for argument 1 from 'SdVolume' to 'FsVolume*'
Multiple libraries were found for "SD.h"
 Used: C:\Program Files (x86)\Arduino\hardware\teensy\avr\libraries\SD
 Not used: C:\Program Files (x86)\Arduino\libraries\SD
Multiple libraries were found for "Audio.h"
 Used: C:\Users\Niels\Documents\Arduino\libraries\Audio
 Not used: C:\Program Files (x86)\Arduino\hardware\teensy\avr\libraries\Audio
no matching function for call to 'SdFile::openRoot(SdVolume&)'

This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.

**Comment by K7MDL2** [comment]
You need to use 1.8.13 and the older teensyduino 1.54 i think it is.

There is a recent thread where someone had the sane problem, the solution was to downgrade. Likely a built in library change.

Sent from my T-Mobile 4G LTE Device
Get Outlook for Android<https://aka.ms/ghei36>
________________________________
From: oz9ny ***@***.***>
Sent: Friday, October 15, 2021 8:48:22 AM
To: K7MDL2/KEITHSDR ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [K7MDL2/KEITHSDR] Can't compile (#3)


Hi
I have tried to compile this project with no success.

My setup: Win10, Arduino 1.8.16, Teensyduino 1.55

Below is the error messages from compilation:

73 de Niels, OZ9NY

Arduino: 1.8.16 (Windows 10), TD: 1.55, Board: "Teensy 4.1, Serial, 600 MHz, Faster, US English"

SDR_CAT.cpp:64: warning: 'msg' defined but not used
static char msg[S_BUFF];
^

SD_Card.cpp: In function 'void SD_CardInfo()':
SD_Card.cpp:177: error: no matching function for call to 'SdFile::openRoot(SdVolume&)'
root.openRoot(volume);
^
In file included from C:\Program Files (x86)\Arduino\hardware\teensy\avr\libraries\SdFat\src/FsLib/FsLib.h:32:0,
from C:\Program Files (x86)\Arduino\hardware\teensy\avr\libraries\SdFat\src/SdFat.h:35,
from C:\Program Files (x86)\Arduino\hardware\teensy\avr\libraries\SD\src/SD.h:27,
from C:\Users\Niels\Documents\hamradio\KEITHSDR\SDR_RA8875\SDR_RA8875.h:19,
from C:\Users\Niels\Documents\hamradio\KEITHSDR\SDR_RA8875\SD_Card.cpp:7:

C:\Program Files (x86)\Arduino\hardware\teensy\avr\libraries\SdFat\src/FsLib/FsFile.h:458:8: note: candidate: bool FsBaseFile::openRoot(FsVolume*)
bool openRoot(FsVolume* vol);
^
C:\Program Files (x86)\Arduino\hardware\teensy\avr\libraries\SdFat\src/FsLib/FsFile.h:458:8: note: no known conversion for argument 1 from 'SdVolume' to 'FsVolume*'
Multiple libraries were found for "SD.h"
Used: C:\Program Files (x86)\Arduino\hardware\teensy\avr\libraries\SD
Not used: C:\Program Files (x86)\Arduino\libraries\SD
Multiple libraries were found for "Audio.h"
Used: C:\Users\Niels\Documents\Arduino\libraries\Audio
Not used: C:\Program Files (x86)\Arduino\hardware\teensy\avr\libraries\Audio
no matching function for call to 'SdFile::openRoot(SdVolume&)'

This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FK7MDL2%2FKEITHSDR%2Fissues%2F3&data=04%7C01%7C%7Cf335924023924635735e08d98fda12e1%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637698989054947856%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=Hdx%2B9i7XDuc6vapWa%2FbzVXt3YmGwKc3XuZhYr71i6RM%3D&reserved=0>, or unsubscribe<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAPLVE7D2FT6ND4YBVSROXCTUHAPJNANCNFSM5GB42AJQ&data=04%7C01%7C%7Cf335924023924635735e08d98fda12e1%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637698989054957812%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=Uzn8NdibXUbnaFPXy3IN0iJJaX%2B05WyHjDQKV%2FtIt7A%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7C%7Cf335924023924635735e08d98fda12e1%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637698989054957812%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=fUadkE3bT%2BWWF4%2B0CFJ83CQ0h6AtpAnw3FqaWojH4PI%3D&reserved=0> or Android<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7C%7Cf335924023924635735e08d98fda12e1%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637698989054967766%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=gJ%2FFxmpO9RPt01If8nfRkyV6D6dXVia5nYeB5I0sg6Y%3D&reserved=0>.

**Comment by oz9ny** [comment]
Thank you. But I could not find an easy way to downgrade teensyduino to a lower version. 
So I began looking into the teensy SD-card examples and they all worked out of the box.
Comparing your void SD_CardInfo(void) routine with teensy example CardInfo I discovered that theese lines were commented out in the examples:

  //Serial.println("\nFiles found on the card (name, date and size in bytes): ");
  //root.openRoot(volume);
  
  // list all files in the card with date and size
  //root.ls(LS_R | LS_DATE | LS_SIZE);

Doing the same in your SD_Card.cpp the project compiles without error.

I have not checked for side effects because my first goal is to see if I could use my Adafruit RA8875 instead of the Buydisplay ra8875 display.

73 de niels

**Comment by K7MDL2** [comment]
The SD card is not really used at this time.  That code was added in to experiment with getting SD card storage o work and all it does today is write the data tables then read it back.  You can bypass it all.

I have not upgraded from 1.8.13 yet, on the to do list for this winter.  I have been playing around with my Teensy SDR this week adding USB MIDI commands to the i2c encoders and push switches as a trial run to make a standalone USB pod with knobs and switches for a SDR apps on a PC or RPi.


Mike
K7MDL  EL87sm & CN88sf

**Comment by K7MDL2** [comment]
I will be resuming development on this project soon.  One of the tasks is to update everything to the latest Arduino and TeensyDuino versions.  The latest Teensy Duino has a lot of changes listed for the the SD card area.

**Comment by K7MDL2** [comment]
Today I updated from Arduino 1.8.13 to 1.8.19 and to TeensyDuino 1.56.  The RA8875UserSettings.h file needs to be edited to ensure the capacitive touchscreen controller is enabled
Line 63   #define USE_FT5206_TOUCH//capacitive touch screen 

I have also commented out the lines in SD_Card and those errors went away.  I need to look into those more, there were TeensyDuino library changes for SDCard.

I can now get a clean compile including experimental USB MIDI controller code I am adding.  I have not tested on hardware yet but the compile looks good.
